File: TeamCode/Lilbro5000/src/main/java/org/firstinspires/ftc/teamcode/MainTeleOp.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

import au.edu.sa.mbhs.studentrobotics.bunyipslib.BunyipsOpMode;
import au.edu.sa.mbhs.studentrobotics.bunyipslib.BunyipsSubsystem;
import au.edu.sa.mbhs.studentrobotics.bunyipslib.util.Geometry;

/**
 * Primary TeleOp.
 * Bootstrapped through bunyipslib-for-rookies.
 *
 * @author Ziya, 2025
 */
@TeleOp(name = "TeleOp")
public class MainTeleOp extends BunyipsOpMode {
    private final Lilbro5000 robot = new Lilbro5000();
    private boolean toggle;

    @Override
    protected void onInit() {
        robot.init();
    }

    @Override
    protected void activeLoop() {
        double forward = -gamepad1.left_stick_y;
        double strafe = -gamepad1.left_stick_x;
        double rotation = -gamepad1.right_stick_x;
        robot.drive.setPower(Geometry.vel(forward, strafe, rotation));

        if (gamepad2.b) {
            robot.intake.setPower(-1);
        } else {
            robot.intake.setPower(gamepad2.left_trigger);
        }

        if (gamepad2.left_bumper) {
            robot.transfer.setPower(1);
        } else if (gamepad2.a) {
            robot.transfer.setPower(-1);
        } else {
            robot.transfer.setPower(0);
        }

        if (gamepad2.rightBumperWasPressed()) {
            toggle = !toggle;
        }
        if (gamepad2.right_trigger == 1) {
            robot.outtake.setPower(1);
        } else if (toggle) {
            robot.outtake.setPower(0.5);
        } else {
            robot.outtake.setPower(0);
        }

        if (gamepad2.x) {
            robot.middletake.setPower(1);
        } else if (gamepad2.y) {
            robot.middletake.setPower(-1);
        } else {
            robot.middletake.setPower(0);
//        }
//        if (gamepad2.left_bumper) {
//            robot.intake.setPower(1);
//            robot.middletake.setPower(-1);
//        } else {
//            robot.intake.setPower(0);
//            robot.middletake.setPower(0);
//        }
//
//        if (gamepad2.right_bumper) {
//            robot.outtake.setPower(1);
//            robot.transfer.setPower(1);
//        } else {
//            robot.outtake.setPower(0);
//            robot.transfer.setPower(0);
        }

        BunyipsSubsystem.updateAll();
    }
}

File: TeamCode/Proto/src/main/java/org/firstinspires/ftc/teamcode/teleop/MainTeleOp.kt
package org.firstinspires.ftc.teamcode.teleop

import au.edu.sa.mbhs.studentrobotics.bunyipslib.CommandBasedBunyipsOpMode
import au.edu.sa.mbhs.studentrobotics.bunyipslib.external.units.Units.Seconds
import au.edu.sa.mbhs.studentrobotics.bunyipslib.tasks.HolonomicDriveTask
import au.edu.sa.mbhs.studentrobotics.bunyipslib.tasks.HolonomicVectorDriveTask
import au.edu.sa.mbhs.studentrobotics.bunyipslib.transforms.Controls
import au.edu.sa.mbhs.studentrobotics.bunyipslib.transforms.Controls.Companion.rising
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.Proto

/**
 * Primary TeleOp for Proto.
 *
 * @author Lucas Bubner, 2024
 */
@TeleOp(name = "TeleOp")
open class MainTeleOp : CommandBasedBunyipsOpMode() {
    override fun assignCommands() {
        HolonomicVectorDriveTask(gamepad1, Proto.drive).setAsDefaultTask()
        driver() whenPressed Controls.BACK run HolonomicDriveTask(gamepad1, Proto.drive) finishIf { gamepad1 rising Controls.BACK }
        Proto.lift.tasks.control { -gamepad2.lsy.toDouble() }.setAsDefaultTask()
        Proto.rotator.tasks.controlDelta { gamepad2.rsy.toDouble() * (timer.deltaTime() to Seconds) }.setAsDefaultTask()
        Proto.intake.tasks.control { if (gamepad2.x) 1.0 else if (gamepad2.y) -1.0 else 0.0 }.setAsDefaultTask()
        operator() whenRising (Controls.Analog.RIGHT_TRIGGER to { v -> v == 1.0f }) run Proto.lift.tasks.home() finishIf { gamepad2.lsy != 0.0f }
    }
}